### PR TITLE
Synapse Support

### LINKED ISSUE #175: Synapse Workspace support
As an engineer, I would like to create a function/cmdlet to support Synapse Workspace checks so that we can support future customers' potential needs.

Extra ask from Lion:  Is it possible to validate ports are open/firewall rules are set? Is this a responsibility of Benchpress?
	
- [ ] Create `Confirm-SynapseWorkspace.ps1` and unit tests
- [ ] Wire up `Confirm-SynapseWorkspace` to `Common.ps1`
- [ ] Create example tests, bicep and documentation in `examples` folder

### PR DESCRIPTION
**Pull request by msha1026**
Support for Synapse Workspace, Spark Pool and Sql Pool

Closes #175, #180, #181

### CODE REVIEW

**Review by marclerwick — CHANGES_REQUESTED** [review]

File: Modules/BenchPress.Azure/Public/Confirm-Resource.ps1 [hunk 1]
```diff
@@ -100,18 +107,20 @@ function Confirm-Resource {
...
     }
 
     $ConfirmResult = Get-ResourceByType @ResourceParams
 
     if ($null -eq $ConfirmResult) {
       Write-Error "Resource not found" -Category InvalidResult -ErrorId "InvalidResource"
       $ConfirmResult = [ConfirmResult]::new($null)
-    } elseif ($ConfirmResult.Success -and -not [string]::IsNullOrWhiteSpace($PropertyKey)) {
+    }
+    elseif ($ConfirmResult.Success -and -not [string]::IsNullOrWhiteSpace($PropertyKey)) {
```
[the remark this answers is not in the document]
Comment (msha1026) [reply Modules/BenchPress.Azure/Public/Confirm-Resource.ps1:123]: Gotcha. My VSCode formatting extension must not be using that practice. Will revert.

File: Modules/BenchPress.Azure/Public/Confirm-Resource.ps1 [hunk 2]
```diff
@@ -91,6 +95,9 @@ function Confirm-Resource {
     [Parameter(Mandatory = $false)]
     [string]$ServerName,
 
+    [Parameter(Mandatory = $false)]
+    [string]$SynapseWorkspaceName,
```
[the remark this answers is not in the document]
Comment (msha1026) [reply Modules/BenchPress.Azure/Public/Confirm-Resource.ps1:99]: Yes good idea. Will apply that change.

**Review by marclerwick — APPROVED** [review]

**Comment by ozboms** [comment]
Could you add tests for `BeDeployed`, `BeInResourceGroup` and `BeInLocation` similar to my current PR? It should be three tests for each of them and just checking if the Synapse resources support them.

**Review by ozboms — CHANGES_REQUESTED** [review]
Up top comment

**Comment by msha1026** [comment]
> Could you add tests for `BeDeployed`, `BeInResourceGroup` and `BeInLocation` similar to my current PR? It should be three tests for each of them and just checking if the Synapse resources support them.

There is no ResourceGroup or ResourceGroupName properties on these resources, so I did not add `BeInResourceGroup`. I added the other two custom assertions though.

**Comment by ozboms** [comment]
> > Could you add tests for `BeDeployed`, `BeInResourceGroup` and `BeInLocation` similar to my current PR? It should be three tests for each of them and just checking if the Synapse resources support them.
> 
> There is no ResourceGroup or ResourceGroupName properties on these resources, so I did not add `BeInResourceGroup`. I added the other two custom assertions though.

There are properties for _some_ of those [resources ](https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.commands.synapse.models.pssynapsesqlpool?view=az-ps-latest)but the current code from main doesn't account for these discrepancies.

I address this in my current PR. So if yours goes in first, I can add them/check on them. If mine goes in first, vice-versa.

**Review by ozboms — APPROVED** [review]

**Pull request merged**